### ISSUE TITLE
[Feature]: Function Calls for models without function calls

### ISSUE THREAD

**Issue by Nivek92** [BerriAI/litellm#557]
### The Feature

Currently when you try to use function calls with models that do not natively support them via API you receive `ValueError: LiteLLM.Exception: Function calling is not supported by this provider`

This is inconvinient e.g. when you use autogen.

To work around that litellm could add the function config to the user prompt.

I believe OpenAI does the same via the API, at least in the beginning they did that, maybe they improved upon that, but for other models it would be a viable approach.

Here's an example of what to add to the user prompt.
```
The following functions are available to you:
{
                "name": "search",
                "description": "google search for relevant information",
                "parameters": {
                    "type": "object",
                    "properties": {
                        "query": {
                            "type": "string",
                            "description": "Google search query",
                        }
                    },
                    "required": ["query"],
                },
            }
```

### Motivation, pitch

Currently when you try to use function calls with models that do not natively support them via API you receive `ValueError: LiteLLM.Exception: Function calling is not supported by this provider`

This is inconvinient e.g. when you use autogen.

**Comment by ghost** [comment]
Hey @Nivek92 we expose 'litellm.drop_params = True' as something you can set to prevent this. 

Are you facing this issue with the proxy? Will put a fix for it there.

**Comment by ghost** [comment]
Re-reading this. I'm wrong. I think you don't want to drop the argument, but pass it as the prompt instead. That's an interesting approach. I'll push something for this (maybe as a configurable Param). 

@Nivek92 im assuming this is via proxy?

**Comment by Nivek92** [comment]
@krrishdholakia yeah I'm using the proxy, but I guess the same happens when using the library itself doesn't it.

What I suggest is in the case when the model api does not allow functions natively, to append or prepend the function schema (json object) to the message or the system prompt before sending it to the model, so that the model knows about the available functions.

**Comment by ghost** [comment]
@Nivek92 pushed an update here - 72ac4fe7817508c20e55f41360a529618986ef4b

tldr; you can now tell litellm to add functions to prompt if the provider doesn't natively support the param (i.e. any non-openai / azure provider) via `litellm.add_function_to_prompt = True`


## code example
```
import litellm 
litellm.add_function_to_prompt = True


model = "claude-instant-1"
messages=[{"role": "user", "content": "what's the weather in sf?"}]
functions = [
    {
    "name": "get_current_weather",
    "description": "Get the current weather in a given location",
    "parameters": {
        "type": "object",
        "properties": {
        "location": {
            "type": "string",
            "description": "The city and state, e.g. San Francisco, CA"
        },
        "unit": {
            "type": "string",
            "enum": ["celsius", "fahrenheit"]
        }
        },
        "required": ["location"]
    }
    }
]
response = litellm.completion(model=model, messages=messages, functions=functions)
print(f'response: {response}')
```

**Comment by ghost** [comment]
will update the ticket once this is in prod + in the proxy

**Comment by ghost** [comment]
This now also in the proxy

**Comment by ghost** [comment]
this is in prod from v`0.6.1`

**Comment by ghost** [comment]
@Nivek92 did this solve your issue? please let me know / re-open the ticket if not
